rust/src/state.rs

```diff
@@ -0,0 +1,95 @@
+#[derive(Debug, Clone)]
```

Comment:
Hmm, I think if this makes it to documentation/example code rather than a demo we could do a bit better. What is the surrounding context of this? Does it end up in our documentation?

Reply:
Tagging @morsapaes  since I'm unsure about the future of this code.

Reply:
The examples in this repo are used as a basis for the existing [integration guides](https://materialize.com/docs/integrations/#libraries-and-drivers) in the documentation (no Rust one in there yet). We plan to make these runnable end-to-end, at some point.